cli/src/utils.rs

```diff
@@ -93,18 +93,16 @@ pub fn check_inferred(got: &[InferenceFact], expected: &[InferenceFact]) -> Trac
     Ok(())
 }
 
-pub fn clarify_tvalues(values: &TVec<TValue>) -> TVec<TValue> {
+pub fn clarify_tvalues(values: &TVec<TValue>) -> TractResult<TVec<TValue>> {
     values
         .iter()
         .map(|t| {
-            t.to_scalar::<Opaque>()
-                .and_then(|ot| ot.clarify_to_tensor().transpose())
-                .ok()
-                .flatten()
+            Ok(t.to_scalar::<Opaque>()?
+                .clarify_to_tensor()?
                 .map(|ot| ot.into_tvalue())
-                .unwrap_or(t.clone())
+                .unwrap_or(t.clone()))
```

Comment:
unwrap_or_else ?

Reply:
mmm... not too bad here as we are just shallow-cloning the tvalue, actually (i wonder if clippy is not going to catch this one anyway)